tests/components/home_connect/conftest.py

```diff
@@ -369,17 +300,78 @@ async def set_program_option_side_effect(ha_id: str, *_, **kwargs) -> None:
             ]
         )
 
+    appliances = [appliance] if appliance else appliances
+
     async def stream_all_events() -> AsyncGenerator[EventMessage]:
         """Mock stream_all_events."""
         while True:
             for event in await event_queue.get():
                 yield event
 
-    mock.get_home_appliances = AsyncMock(return_value=copy.deepcopy(MOCK_APPLIANCES))
+    mock.get_home_appliances = AsyncMock(return_value=ArrayOfHomeAppliances(appliances))
+
+    def _get_specific_appliance_side_effect(ha_id: str) -> HomeAppliance:
+        """Get specific appliance side effect."""
+        for appliance_ in appliances:
+            if appliance_.ha_id == ha_id:
+                return appliance_
+        raise HomeConnectApiError("error.key", "error description")
+
     mock.get_specific_appliance = AsyncMock(
         side_effect=_get_specific_appliance_side_effect
     )
     mock.stream_all_events = stream_all_events
+
+    async def _get_all_programs_side_effect(ha_id: str) -> ArrayOfPrograms:
```

Comment:
This block is just moved down from above and nested into this function, to have access to `appliances`.